token/program-2022/tests/serialization.rs

```diff
@@ -35,3 +52,115 @@ fn token_program_serde_with_none() {
+    assert_eq!(&serialized, serialized_expected);
+
+    let deserialized = serde_json::from_str::<confidential_transfer::instruction::InitializeMintData>(&serialized).unwrap();
+    assert_eq!(inst, deserialized);
+}
+
+
+#[test]
+// tests serde of ix containing OptionalNonZeroPubkey, PodBool and OptionalNonZeroElGamalPubkey with
+// null values
+fn token_program_extension_serde_with_none() {
+    let authority: OptionalNonZeroPubkey = None.try_into().unwrap();
+
+    let auditor_elgamal_pubkey: OptionalNonZeroElGamalPubkey = OptionalNonZeroElGamalPubkey::default();
+
+    let inst = confidential_transfer::instruction::InitializeMintData {
+        authority,
+        auto_approve_new_accounts: false.into(),
+        auditor_elgamal_pubkey,
+    };
+
+    let serialized = serde_json::to_string(&inst).unwrap();
+    let serialized_expected = &format!("{{\"authority\":null,\"auto_approve_new_accounts\":false,\"auditor_elgamal_pubkey\":null}}");
+    assert_eq!(&serialized, serialized_expected);
+
+    let deserialized = serde_json::from_str::<confidential_transfer::instruction::InitializeMintData>(&serialized_expected).unwrap();
+    assert_eq!(inst, deserialized);
+}
+
+#[test]
+// tests serde of ix containing DecryptableBalance and PodU64
+fn token_program_extension_serde_config_account() {
+    let decryptable_zero_balance = AeCiphertext([56, 22, 102, 48, 112, 106, 58, 25, 25, 244, 194,
+        217, 73, 137, 73, 38, 24, 26, 36, 25, 235, 234, 68, 181, 11, 82, 170, 163, 89, 205, 113,
+        160, 55, 16, 35, 151]);
+
+    let inst = confidential_transfer::instruction::ConfigureAccountInstructionData {
+        decryptable_zero_balance,
+        maximum_pending_balance_credit_counter: 1099.into(),
+        proof_instruction_offset: 100,
+    };
+
+    let serialized = serde_json::to_string(&inst).unwrap();
+    let serialized_expected = &format!("{{\"decryptable_zero_balance\":\"OBZmMHBqOhkZ9MLZSYlJJhgaJBnr6kS1C1Kqo1nNcaA3ECOX\",\"maximum_pending_balance_credit_counter\":1099,\"proof_instruction_offset\":100}}");
+    assert_eq!(&serialized, serialized_expected);
+
+    let deserialized = serde_json::from_str::<confidential_transfer::instruction::ConfigureAccountInstructionData>(&serialized_expected).unwrap();
+    assert_eq!(inst, deserialized);
+}
+
+#[test]
+// tests serde of ix containing pod::ElGamalPubkey
+fn token_program_extension_serde_config_data() {
+    use spl_token_2022::extension::confidential_transfer_fee::{
+        instruction::InitializeConfidentialTransferFeeConfigData
+    };
+
+    let withdraw_withheld_authority_elgamal_pubkey = ElGamalPubkey([162, 23, 108, 36, 130, 143,
+        18, 219, 196, 134, 242, 145, 179, 49, 229, 193, 74, 64, 3, 158, 68, 235, 124, 88, 247, 144,
+        164, 254, 228, 12, 173, 85]);
+
+    let inst = InitializeConfidentialTransferFeeConfigData {
+        authority: Default::default(),
```

Comment:
nit: can you avoid `Default::default()` and spell out the type?
```suggestion
        authority: OptionalNonZeroPubkey::default(),
```

Reply:
Done